src/celeritas/em/msc/UrbanMscScatter.hh

```diff
@@ -222,12 +324,14 @@ CELER_FUNCTION auto UrbanMscScatter::operator()(Engine& rng) -> MscInteraction
     MscInteraction result;
     result.action = MscInteraction::Action::scattered;
     {
-        // This should only be needed to silence compiler warning
+        // This should only be needed to silence compiler warning, since the
+        // displacement should be ignored since our action result is
+        // 'scattered'
         result.displacement = {0, 0, 0};
     }
 
     // Calculate displacement
-    if (is_displaced_ && tau_ >= params_.tau_small)
```

Comment:
So, is the second condition absorbed into is_displaced? Why?

Reply:
This is equivalent to `tau_ < params_.tau_small` forcing `is_displaced_` to be `false`. This condition is now precalculated in the constructor because it changes the whole sampling strategy. (It's not affected by the random number selection so ideally shouldn't be inside the `operator()`.